### DISCUSSION TITLE
Best practice for adopting OpenAPI codegen when current codebase is all reactive/mutiny?

### DISCUSSION THREAD

**Discussion by hiett** [quarkusio/quarkus#30641] [Q&A]
Hey all,

I've got to the point of having a number of microservices in different languages, many of which in Quarkus. Right now, all of the Quarkus services are using RestEasy Reactive with Mutiny, and the services are all coded to be reactive-first (that is, there is a lot of Mutiny code, and everything uses Quarkus-provided libraries that provide me Unis).

I have been generating OpenAPI schemas from my Quarkus services, and using them with great success in the other languages and frameworks. However, the Quarkus codegen I tried (quarkus-openapi-generator) doesn't yet have Resteasy Reactive support (https://github.com/quarkiverse/quarkus-openapi-generator/issues/150).

Right now, to get around this, I have been manually defining the clients. Obviously the only big difference is I've been adding in some Uni's here and there. However, this has gotten to the point of not being very maintainable, and I'd really like to use some sort of codegen for all these OpenAPI schemas I have.

So here's my question:

Should I keep going as I am in doing it by-hand until the quarkus-openapi-generator supports Reactive out of the box?
...or...
What would be the best practice in making the API calls to get them back as Unis and apart of the IO threads? Would I have to write some sort of wrapper for each API call? If so, what's the best practice for doing this?
...or...
Is there some quick Quarkus way of solving this problem that I've missed?

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
I'm not aware of such a generator. But @EricWittmann is more likely to know the definitive answer.

**Comment by ghost** [comment]
HeyHey, are there any updates on this topic in 2024? 
We are facing the same issue like @hiett. We will probably go for auto generated DTOs from the OpenAPI spec and use them in order to create the endpoints manually with rest reactive. 
Is there even a way to participate in the development of mutiny support?

**Comment by agreedSkiing** [comment]
Quarkus-openapi-generator and the offical openapi-generator seem to have support for this now. Docs from quarkus-openapi-generator https://docs.quarkiverse.io/quarkus-openapi-generator/dev/server.html and openapi-generator https://github.com/OpenAPITools/openapi-generator/pull/11554 pr
@wielandpaul @hiett
